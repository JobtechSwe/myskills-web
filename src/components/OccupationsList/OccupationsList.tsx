import React from 'react'
import List from '../List'
import { OntologyType, OntologyConceptResponse } from '../../generated/myskills'

type MutationVariables = {
  variables: {
    occupation: {
      name: string
      id: string
      type: OntologyType
    }
  }
}

interface ExperienceListProps {
  occupations: OntologyConceptResponse[]
  addOccupation?: (variables: MutationVariables) => void
}

const ExperienceList: React.FC<ExperienceListProps> = ({
  occupations,
  addOccupation,
}) => {
  return (
    <List>
      {occupations.map((occupation: OntologyConceptResponse, i: number) => {
        return (
          <li key={i}>
            {addOccupation && (
              <button
                onClick={() =>
                  addOccupation({
                    variables: {
                      occupation,
                    },
                  })
                }
              >
                {occupation.name}
              </button>
            )}
          </li>
        )
      })}
    </List>
  )
}

export default ExperienceList
