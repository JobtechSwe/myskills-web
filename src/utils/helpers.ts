import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
