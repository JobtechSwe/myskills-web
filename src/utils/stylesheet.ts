const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
