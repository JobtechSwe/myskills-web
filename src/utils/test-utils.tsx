import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
