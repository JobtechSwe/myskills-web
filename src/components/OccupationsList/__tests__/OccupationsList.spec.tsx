import * as React from 'react'
import OccupationsList from '../OccupationsList'
import { render } from '../../../utils/test-utils'
import { wait, fireEvent } from 'react-testing-library'

describe('views/OccupationsList', () => {
  let occupations: any
  let addOccupation: any

  beforeEach(() => {
    addOccupation = jest.fn()
    occupations = [
      {
        name: 'Javascript',
        id: 'abc',
      },
      {
        name: 'Java',
        id: 'cda',
      },
    ]
  })

  it('renders empty list', async () => {
    const { container } = render(<OccupationsList occupations={[]} />)

    await wait()

    expect(container).toMatchSnapshot()
  })

  it('renders experineces list', async () => {
    const { container } = render(<OccupationsList occupations={occupations} />)

    await wait()

    expect(container).toMatchSnapshot()
  })

  it('renders experineces list with button', async () => {
    const { container } = render(
      <OccupationsList
        addExperience={addOccupation}
        occupations={occupations}
      />
    )

    await wait()

    expect(container).toMatchSnapshot()
  })

  it('should render with taxonomy query result', async () => {
    const { getByText } = render(
      <OccupationsList
        addOccupation={addOccupation}
        occupations={occupations}
      />
    )

    fireEvent.click(getByText(/java/i))

    await wait()

    expect(addOccupation).toHaveBeenCalledWith({
      variables: {
        occupation: { name: 'Javascript', id: 'abc' },
      },
    })
  })
})
