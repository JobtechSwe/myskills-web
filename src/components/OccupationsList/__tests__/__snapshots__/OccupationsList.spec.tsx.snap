// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/OccupationsList renders empty list 1`] = `
<div>
  <ul
    class="css-1qkqrd8"
    color="blue"
  />
</div>
`;

exports[`views/OccupationsList renders experineces list 1`] = `
<div>
  <ul
    class="css-1qkqrd8"
    color="blue"
  >
    <li
      class="css-6e1t7r"
      color="black"
    >
      Javascript
    </li>
    <li
      class="css-6e1t7r"
      color="black"
    >
      Java
    </li>
  </ul>
</div>
`;

exports[`views/OccupationsList renders experineces list with button 1`] = `
<div>
  <ul
    class="css-1qkqrd8"
    color="blue"
  >
    <li
      class="css-6e1t7r"
      color="black"
    >
      Javascript
    </li>
    <li
      class="css-6e1t7r"
      color="black"
    >
      Java
    </li>
  </ul>
</div>
`;
